### PR TITLE
fix(monitor-setup): stop using `mikefarah/yq` docker

### PR DESCRIPTION
**Pull request by fruch**
we need to edit a remote file, an we used a docker image of yq for that.

changing it to use SCT `remote_file` context manager

Fixes: #6743

### Testing
- [x] - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-5gb-1h-GrowShrinkClusterNemesis-aws-test/2/

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -5573,10 +5572,9 @@ def reconfigure_scylla_monitoring(self):
                 python3 genconfig.py -s -n -d {self.monitoring_conf_dir} {monitoring_targets}
             """), verbose=True)
 
-            if cluster_backend != "docker":
-                node.remoter.sudo(f"docker run --rm -v {self.monitoring_conf_dir}:/workdir"
-                                  f" mikefarah/yq:3 yq w -i node_exporter_servers.yml"
-                                  f" '[0].targets[+]' ''[{node.private_ip_address}]:9100''", verbose=True)
+            with node._remote_yaml(f'{self.monitoring_conf_dir}/node_exporter_servers.yml') as exporter_yaml:  # pylint: disable=protected-access
```
Comment (soyacz) [inline sdcm/cluster.py:5575]: did you test it on docker backend?

**Review by soyacz — COMMENTED** [review]
now, when we don't need to make exception for docker (or any other backend), why not just to updtate `monitoring_targets` and remove whole idea of updating `node_exproter_servers.yml` file?

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py:5575]: doing the docker provision test now

Comment (fruch) [reply sdcm/cluster.py:5575]: but it's only for case of adding new node, and so it should be tested directly

Comment (soyacz) [reply sdcm/cluster.py:5575]: `reconfigure_scylla_monitoring` doesn't take node param, it reconfigures everything.

Comment (fruch) [reply sdcm/cluster.py:5575]: There is a loop on the nodes

Comment (soyacz) [reply sdcm/cluster.py:5575]: monitoring nodes

Comment (soyacz) [reply sdcm/cluster.py:5575]: monitoring nodes

Comment (fruch) [reply sdcm/cluster.py:5575]: It's adding the node exporter of the monitor

**Comment by fruch** [comment]
> now, when we don't need to make exception for docker (or any other backend), why not just to updtate `monitoring_targets` and remove whole idea of updating `node_exproter_servers.yml` file?

`monitoring_targets` is a list we pass on for the db nodes.

adding to that might not exactly get us the result we expect

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py:5575]: no, but we are using that code multiple times on scylla yaml, should work as well.

**Review by vponomaryov — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]

**Pull request merged**